### PR TITLE
renovate: add auto-approve bot for renovate PRs

### PR DESCRIPTION
**Pull request by aanm**
Enable the GitHub "auto-merge" feature in the repository settings at `https://github.com/<org>/<repo>/settings`. If Renovate detects this feature, it allows PRs to be auto-merged by GitHub.

GitHub will auto-merge a PR if all required checks pass and CODEOWNERS have reviewed it. If these conditions are unmet, GitHub won't merge the PR.

To allow Renovate to auto-approve its own PRs, configure Renovate to request a review from the bot `ciliumbot` for PRs with trusted dependencies. The `reviewers` configuration in Renovate will ensure `ciliumbot` is the sole reviewer of Renovate's PRs.

Create a GitHub Action triggered by a review request event, ensuring the PR review was requested by the Renovate bot, the PR was created by Renovate, and the review request is for `ciliumbot`.

Ensure `ciliumbot` belongs to some teams of the CODEOWNERS file but is not auto-assigned reviews by GitHub. This setup allows `ciliumbot` to provide the necessary approvals without manual intervention, enabling seamless integration of Renovate to auto-approve PRs.

The teams that `ciliumbot` will belong to are the ones that usually are selected to review renovate PRs when a trusted dependency is updated.

### CODE REVIEW

File: .github/workflows/auto-approve.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    # Avoid running the 'auto-approve' environment if we don't need to.
+    name: Pre-Approve
+    runs-on: ubuntu-latest
+    if: ${{
+         github.event.pull_request.user.login == 'cilium-renovate[bot]' &&
+         github.triggering_actor == 'cilium-renovate[bot]' &&
+         github.event.requested_reviewer.login == 'ciliumbot'
+        }}
+    steps:
+    - name: Debug
```
Comment (viktor-kurchenko) [inline .github/workflows/auto-approve.yaml:19]: Are you going to remove Debug steps?

**Review by viktor-kurchenko — APPROVED** [review]
LGTM

**Review by bimmlerd — APPROVED** [review]
nice, thanks

File: .github/workflows/auto-approve.yaml (hunk 1 above)
Comment (aanm) [reply .github/workflows/auto-approve.yaml:19]: No, the "steps" field is mandatory on a job so it doesn't hurt to leave them here.

Comment (viktor-kurchenko) [reply .github/workflows/auto-approve.yaml:19]: Gotcha, thanks!

**Pull request merged**